src/routes/chat.router.ts

```diff
@@ -1,14 +1,20 @@
 import express from 'express';
 import { chatController } from '../controllers/chat.controller';
 import { authenticateToken } from '../application/middlewares/authenticateToken';
+import { 
+  validateGetChatRoute,
+  validateDeleteChatRoute,
+  validateUpdateChatRoute,
+  validateViewChatRoute
+} from '../security/requestChatRouteValidator'
 
 const chatRouter = express.Router();
 
-chatRouter.get('/:id', authenticateToken, chatController.getById);
+chatRouter.get('/:id', validateGetChatRoute(), authenticateToken, chatController.getById);
```

Comment:
Change order of middlewares, to authenticate users first, then validate request

Reply:
Is it working with `functionName()` instead of `functionName`? look at `authenticationToken` function

So please change signature of function to be
```
function functionName(req: Request, res: Response): void {
...
}
```